Automattic/jetpack
# Related Posts: Fix context for use with blocks

Fixes #30673 

## Proposed changes:


Adds a second "block context" for use in block rendering. The existing context is still generated for use by the non-block version of the output. With the new context, we also properly set up a link to the category, tag, or comments as appropriate.

With the new context in place, the rendered block should show…

1. Linked up category.
2. Linked up tag.

<img width="684" alt="SCR-20230512-ssje" src="https://github.com/Automattic/jetpack/assets/40267301/db1168a5-d4ba-4323-a7f9-85f13c39b8eb">

In the block editor we should see the same:

<img width="694" alt="SCR-20230512-stda" src="https://github.com/Automattic/jetpack/assets/40267301/b9ae51ed-a798-468d-a6f5-5c86f522dc0d">

### Other information:


## Jetpack product discussion



pejTkB-yT-p2

## Does this pull request change what data or activity we track or use?



No

## Testing instructions:






1. Spin up a test site with this branch.
3. Make sure Related Posts are enabled via Jetpack → Settings → Traffic → Related Posts.
4. Create a new post and add a Related Posts block.
5. Confirm the context (category or tag) is rendered correctly in the editor and on the front-facing page (as annotated in the screenshots). You may need to publish before the placeholders are replaced with actual post content.
